### PR TITLE
[1063] Document storage [2nd PR]

### PR DESCRIPTION
**Pull request by Sybrand**
* Refactoring lots of code out into "wells" from "gwells" app (working in progress).
* Vue component to display additional documents (private and public).
* Fix to keycloak login to stop page reloads.
* Document api (minio) and tests added.
* Additional permissions for viewing private documents.

### CODE REVIEW

**Comment by Sybrand** [comment]
[EPIC#WellSummary | Add links to internal-only documents |5| !!! [business value ranking:25 ]](https://trello.com/c/IzDBNuCY/768-epicwellsummary-add-links-to-internal-only-documents-5-business-value-ranking25)

File: openshift/backend.dc.json [hunk 1]
```diff
@@ -192,16 +192,6 @@
                                     }
                                 ]
                             }
-                        },
-                        "mid": {
-                            "failurePolicy": "Abort",
-                            "execNewPod": {
-                                "command": [
-                                    "/usr/bin/container-entrypoint",
```
Comment (stephenhillier) [inline openshift/backend.dc.json]: does this not need to be executed anymore?

File: Jenkinsfile [hunk 2]
```diff
@@ -427,6 +427,24 @@ for(String envKeyName: context.env.keySet() as String[]){
...
+                String projectName=context.deployments[envKeyName].projectName
+                String deploymentConfigName="gwells${context.deployments[envKeyName].dcSuffix}"
+                echo "POST-DEPLOY: env:${context.env[envKeyName]}"
+                echo "POST-DEPLOY: deployment:${context.deployments[envKeyName]}"
+                echo "POST-DEPLOY: projectName:${projectName}"
+                echo "POST-DEPLOY: deploymentConfigName:${deploymentConfigName}"
+                openshift.withProject(projectName){
+                    podName=openshift.selector('pod', ['deploymentconfig':deploymentConfigName]).objects()[0].metadata.name
+                }
+                sh "oc exec '${podName}' -n moe-gwells-dev -- bash -c 'cd /opt/app-root/src/scripts/ && pwd && ./post-deploy.sh'"
```
Comment (stephenhillier) [inline Jenkinsfile]: there doesn't seem to be a migration step anymore

Comment (DerekRoberts) [reply Jenkinsfile]: That's been addressed, but pushed to a separate PR.  Good catch!

File: openshift/backend.dc.json (hunk 1 above)
Comment (DerekRoberts) [reply openshift/backend.dc.json]: No, replaced with something in the Jenkinsfile.

**Pull request merged**